From 3e3c3f66b9e9174016cabed472130c5efbd3de43 Mon Sep 17 00:00:00 2001
From: demostanis worlds <demostanis@protonmail.com>
Date: Thu, 23 Nov 2023 04:02:23 +0100
Subject: [PATCH] Add more margin and don't show minimized clients

---
 init.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init.lua b/init.lua
index 1e84a30..4b4edff 100644
--- a/init.lua
+++ b/init.lua
@@ -118,7 +118,7 @@ function _M.getClients()
 
 
 			if addToTable then
-				table.insert(clients, c)
+				--table.insert(clients, c)
 			end
 		end
 	end
@@ -254,7 +254,7 @@ function _M.preview()
 
 	local x = screen[mouse.screen].geometry.x - _M.preview_wbox.border_width
 	local y = screen[mouse.screen].geometry.y + (screen[mouse.screen].geometry.height - h - textboxHeight) / 2
-	_M.preview_wbox:geometry({x = x, y = y, width = W, height = h + textboxHeight})
+	_M.preview_wbox:geometry({x = x, y = y, width = W, height = h + textboxHeight+10})
 
 	-- create a list that holds the clients to preview, from left to right
 	local leftRightTab = {}
-- 
2.42.1

