From 867a108d11b9a654d888ba9d24573435a82b78de Mon Sep 17 00:00:00 2001
From: demostanis <demostanis@protonmail.com>
Date: Thu, 23 Jan 2025 17:12:23 +0100
Subject: [PATCH] push new windows in the stack after the focused window in

---
 objects/client.c | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/objects/client.c b/objects/client.c
index 17ad387..272f3e3 100644
--- a/objects/client.c
+++ b/objects/client.c
@@ -2197,9 +2197,26 @@ client_manage(xcb_window_t w, xcb_get_geometry_reply_t *wgeom, xcb_get_window_at
                          XCB_CONFIG_WINDOW_STACK_MODE,
                          (uint32_t[]) { XCB_STACK_MODE_BELOW});
 
-    /* Duplicate client and push it in client list */
     lua_pushvalue(L, -1);
-    client_array_push(&globalconf.clients, luaA_object_ref(L, -1));
+    /* Push it after the focused client */
+    client_t *cc = luaA_object_ref(L, -1);
+    if (globalconf.focus.client)
+    {
+        int i, n = 1;
+        // indexof doesnt work...
+        foreach(client, globalconf.clients)
+        {
+            if (*client == globalconf.focus.client)
+            {
+                i = n;
+                break;
+            }
+            n++;
+        }
+        client_array_splice(&globalconf.clients, i-1, 0, &cc, 1);
+    }
+    else
+        client_array_push(&globalconf.clients, cc);
 
     /* Set the right screen */
     screen_client_moveto(c, screen_getbycoord(wgeom->x, wgeom->y), false);
-- 
2.48.0

