From 76d383bc5acf530617f0b5fcce8df6eb33395b30 Mon Sep 17 00:00:00 2001
From: demostanis <demostanis@protonmail.com>
Date: Tue, 4 Jul 2023 15:19:57 +0200
Subject: [PATCH] Remove border

---
 src/widgets/capture/overlaymessage.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/widgets/capture/overlaymessage.cpp b/src/widgets/capture/overlaymessage.cpp
index d6f8039..54635ec 100644
--- a/src/widgets/capture/overlaymessage.cpp
+++ b/src/widgets/capture/overlaymessage.cpp
@@ -25,9 +25,6 @@ OverlayMessage::OverlayMessage(QWidget* parent, const QRect& targetArea)
     int opacity = ConfigHandler().contrastOpacity();
     m_textColor =
       (ColorUtils::colorIsDark(m_fillColor) ? Qt::white : Qt::black);
-    // map a background opacity range 0-255 to a fill opacity range 190-160
-    // we do this because an opaque background makes the box look opaque too
-    m_fillColor.setAlpha(160 + (180 - 220) / (255.0 - 0) * (opacity - 255));
     setStyleSheet(
       QStringLiteral("QLabel { color: %1; }").arg(m_textColor.name()));
 
@@ -103,7 +100,7 @@ void OverlayMessage::paintEvent(QPaintEvent* e)
     painter.setRenderHint(QPainter::Antialiasing);
 
     painter.setBrush(QBrush(m_fillColor, Qt::SolidPattern));
-    painter.setPen(QPen(m_textColor, 1.5));
+    painter.setPen(QPen(m_fillColor, 0));
     float margin = painter.pen().widthF();
     painter.drawRoundedRect(
       rect() - QMarginsF(margin, margin, margin, margin), 5, 5);
-- 
2.41.0

