From 987367111de0cca4d8ca180ffc9d2d43984ca9cc Mon Sep 17 00:00:00 2001
From: demostanis <demostanis@protonmail.com>
Date: Thu, 9 May 2024 16:37:23 +0200
Subject: [PATCH] Remove SSL pinning

---
 riseup_vpn_configurator/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/riseup_vpn_configurator/__init__.py b/riseup_vpn_configurator/__init__.py
index 4e33ab8..f41488d 100644
--- a/riseup_vpn_configurator/__init__.py
+++ b/riseup_vpn_configurator/__init__.py
@@ -85,7 +85,7 @@ def update_gateways() -> None:
     #cache_api_ca_cert()
     logging.debug(f"Fetching gateways from {GATEWAYS_API_URL}")
     try:
-        resp = requests.get(GATEWAYS_API_URL, verify=str(api_ca_cert_file))
+        resp = requests.get(GATEWAYS_API_URL)
         gateway_json.write_text(resp.text)
     except Exception as e:
         logging.error(e)
-- 
2.45.0

