From 58c1c587fcf6dba68a085ecfd980f49ccb350118 Mon Sep 17 00:00:00 2001
From: demostanis worlds <demostanis@protonmail.com>
Date: Wed, 4 Oct 2023 17:31:51 +0200
Subject: [PATCH 1/2] do not turn on the front leds

---
 keyboards/planck/ez/ez.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/keyboards/planck/ez/ez.c b/keyboards/planck/ez/ez.c
index 93a17be293..d406a1beba 100644
--- a/keyboards/planck/ez/ez.c
+++ b/keyboards/planck/ez/ez.c
@@ -223,21 +223,7 @@ layer_state_t layer_state_set_kb(layer_state_t state) {
     planck_ez_left_led_off();
     planck_ez_right_led_off();
     state = layer_state_set_user(state);
-    uint8_t layer = get_highest_layer(state);
-    switch (layer) {
-        case 1:
-            planck_ez_left_led_on();
-            break;
-        case 2:
-            planck_ez_right_led_on();
-            break;
-        case 3:
-            planck_ez_right_led_on();
-            planck_ez_left_led_on();
-            break;
-        default:
-            break;
-    }
+    /* do not turn on the front leds */
     return state;
 }
 
-- 
2.42.0

